oberon-chat/chat-client
# Add message starring feature

### Why: 
We would like users to be able to mark messages to refer to later.

### What: 
This PR adds the UI component that allows users to click on messages they would like to save for later reference. This UI component will communicate with chat-server to add the user and message pair to the `starred_messages` table. https://github.com/oberon-chat/chat-server/pull/1 provides the supporting backend logic.

### Verification Steps
It will be easiest to verify both this PR and https://github.com/oberon-chat/chat-server/pull/1 simultaneously.

- [x] Pull down each branch locally

- [x] Run migrations for https://github.com/oberon-chat/chat-server/pull/1

- [x] Open a terminal window that you can run database queries in on the server side. Query `ChatServer.Repo.all(ChatServer.Schema.StarredMessage)` and note that there are no StarredMessage records

- [x] Run the client and server and navigate to chat.dev

- [x] Sign in as any type of user, enter a chat room. Hover over the messages in the room and note that each has a empty star. 

- [x]  Before clicking on any stars, note the empty value for starred_messages in your Redux state.

- [x] Click on a couple messages, note that your redux state has updated to include those messages.

- [x] Navigate to the terminal window you made your previous `ChatServer.Repo.all(ChatServer.Schema.StarredMessage)` query. Run the query again and note that the messages you just starred are present (you should not need to recompile in order to see these records).

- [x] Unstar one of the messages. Note that it has been deleted both from your Redux state and from the database.

- [x] Leaving some messages starred. Log out and delete your cache (or use a private window). Log back in with the same user and note that the messages you previously starred are still starred.
